### DISCUSSION TITLE
How to use this module with standard firebase and plugins

### DISCUSSION THREAD

**Discussion by Lucien950** [nuxt-community/firebase-module#450] [General]
This module can work with nuxt and related, but how does it work with other files? With standard firebase, I can just import 'firebase' but I can't just pull this.$fire without context

**Comment by lupas** [comment]
@Lucien950 
Unfortunately, there's currently no easy way for this other than passing the context with each function call you make in your modules.

**Comment by Lucien950** [comment]
How about if you need to make a type with firebase, but there is no parameter to pass the context into?

**Comment by Lucien950** [comment]
say for example you need to use a $fireModule type for typescript, but you don't have the context

**Reply by lupas** [reply]
Does this work?

```js
import type firebase from 'firebase'

export default function (fireModule: typeof firebase) {
  // fireModule is now typed
  console.log(fireModule.functions)
}
```
